actors/builtin/market/market_test.go

```diff
@@ -693,6 +695,363 @@ func TestPublishStorageDealsFailures(t *testing.T) {
 	}
 }
 
+func TestActivateDeals(t *testing.T) {
+	marketActor := tutil.NewIDAddr(t, 100)
+	owner := tutil.NewIDAddr(t, 101)
+	provider := tutil.NewIDAddr(t, 102)
+	worker := tutil.NewIDAddr(t, 103)
+	client := tutil.NewIDAddr(t, 104)
+	mAddrs := &minerAddrs{owner, worker, provider}
+
+	startEpoch := abi.ChainEpoch(10)
+	endEpoch := abi.ChainEpoch(20)
+	currentEpoch := abi.ChainEpoch(5)
+	sectorExpiry := abi.ChainEpoch(100)
+
+	t.Run("active deals multiple times with different providers", func(t *testing.T) {
+		rt, actor := basicMarketSetup(t, marketActor, owner, provider, worker, client)
+		rt.SetEpoch(currentEpoch)
+
+		// provider 1 publishes deals1 and deals2 and deal3
+		dealId1 := actor.generateAndPublishDeal(rt, client, mAddrs, startEpoch, endEpoch)
+		dealId2 := actor.generateAndPublishDeal(rt, client, mAddrs, startEpoch, endEpoch+1)
+		dealId3 := actor.generateAndPublishDeal(rt, client, mAddrs, startEpoch, endEpoch+2)
+
+		// provider2 publishes deal4 and deal5
+		provider2 := tutil.NewIDAddr(t, 401)
+		mAddrs.provider = provider2
+		dealId4 := actor.generateAndPublishDeal(rt, client, mAddrs, startEpoch, endEpoch)
+		dealId5 := actor.generateAndPublishDeal(rt, client, mAddrs, startEpoch, endEpoch+1)
+
+		// provider1 activates deal 1 and deal2 but that does not activate deal3 to deal5
+		actor.activateDeals(rt, sectorExpiry, provider, currentEpoch, dealId1, dealId2)
+		actor.assertDealsNotActivated(rt, currentEpoch, dealId3, dealId4, dealId5)
+
+		// provider3 activates deal5 but that does not activate deal3 or deal4
+		actor.activateDeals(rt, sectorExpiry, provider2, currentEpoch, dealId5)
+		actor.assertDealsNotActivated(rt, currentEpoch, dealId3, dealId4)
+
+		// provider1 activates deal3
+		actor.activateDeals(rt, sectorExpiry, provider, currentEpoch, dealId3)
+		actor.assertDealsNotActivated(rt, currentEpoch, dealId4)
+	})
+}
+
+func TestActivateDealFailures(t *testing.T) {
+	marketActor := tutil.NewIDAddr(t, 100)
+	owner := tutil.NewIDAddr(t, 101)
+	provider := tutil.NewIDAddr(t, 102)
+	worker := tutil.NewIDAddr(t, 103)
+	client := tutil.NewIDAddr(t, 104)
+	mAddrs := &minerAddrs{owner, worker, provider}
+
+	startEpoch := abi.ChainEpoch(10)
+	endEpoch := abi.ChainEpoch(20)
+	sectorExpiry := abi.ChainEpoch(100)
+
+	// caller is not the provider
+	{
+		t.Run("fail when caller is not the provider of the deal", func(t *testing.T) {
+			rt, actor := basicMarketSetup(t, marketActor, owner, provider, worker, client)
+			provider2 := tutil.NewIDAddr(t, 201)
+			mAddrs2 := &minerAddrs{owner, worker, provider2}
```

Comment:
I think it's equivalent, but this might be simpler if the deal uses the standard `mAddrs` and just set a different address at 763.